apps/api-extractor/src/api/model/ApiModel.ts

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import { ApiItem, ApiItemKind } from './ApiItem';
+import { ApiItemContainerMixin } from '../mixins/ApiItemContainerMixin';
+import { ApiPackage } from './ApiPackage';
+import { PackageName } from '@microsoft/node-core-library';
+import { DeclarationReferenceResolver, IResolveDeclarationReferenceResult } from './DeclarationReferenceResolver';
+import { DocDeclarationReference } from '@microsoft/tsdoc';
+
+/** @public */
+export class ApiModel extends ApiItemContainerMixin(ApiItem) {
+  private readonly _resolver: DeclarationReferenceResolver;
+
+  private _packagesByName: Map<string, ApiPackage> | undefined = undefined;
+
+  public constructor() {
+    super({ name: 'MODEL' });
+
+    this._resolver = new DeclarationReferenceResolver(this);
+  }
+
+  public loadPackage(apiJsonFilename: string): ApiPackage {
+    const apiPackage: ApiPackage = ApiPackage.loadFromJsonFile(apiJsonFilename);
+    this.addMember(apiPackage);
+    return apiPackage;
+  }
+
+  /** @override */
+  public get kind(): ApiItemKind {
+    return ApiItemKind.Model;
+  }
+
+  /** @override */
+  public get canonicalReference(): string {
+    return this.name;
+  }
+
+  public get packages(): ReadonlyArray<ApiPackage> {
+    return this.members as ReadonlyArray<ApiPackage>;
+  }
+
+  /** @override */
+  public addMember(member: ApiPackage): void {
+    if (member.kind !== ApiItemKind.Package) {
+      throw new Error('Only items of type ApiPackage may be added to an ApiModel');
+    }
+    super.addMember(member);
+
+    this._packagesByName = undefined; // invalidate the cache
+  }
+
+  /**
+   * Efficiently finds a package by the NPM package name.
+   *
+   * @remarks
+   *
+   * If the NPM scope is omitted in the package name, it will still be found provided that it is an unambiguous match.
```

Comment:
>If the NPM scope is omitted in the package name, it will still be found provided that it is an unambiguous match. [](start = 5, length = 113)

Why is this useful?

Reply:
It's super convenient to write this:

```
/** See {@link sp-core-library#Guid} */
```

instead of this:

```
/** See {@link @microsoft/sp-core-library#Guid} */
```

I'll explain that better

---
In reply to: [236469729](https://github.com/Microsoft/web-build-tools/pull/952#discussion_r236469729) [](ancestors = 236469729)